otx/core/data/adapter/anomaly_dataset_adapter.py

```
@@ -0,0 +1,240 @@
+
+
+class AnomalyBaseDatasetAdapter(BaseDatasetAdapter):
+    """BaseDataset Adpater for Anomaly tasks inherited from BaseDatasetAdapter."""
+    def __init__(
+        self,
+        task_type: TaskType,
+        train_data_roots: str = None,
+        val_data_roots: str = None,
+        test_data_roots: str = None,
+        unlabeled_data_roots: str = None,
+    ):
+        self.task_type = task_type
+        self.domain = task_type.domain
+        self.data_type = None  # type: Any
+        self.is_train_phase = None  # type: Any
+
+        self.dataset = self._import_dataset(
+            train_data_roots=train_data_roots,
+            val_data_roots=val_data_roots,
+            test_data_roots=test_data_roots,
+            unlabeled_data_roots=unlabeled_data_roots
+        )
+
+        self.label_schema = None # type: LabelSchemaEntity
+
+    def _import_dataset(
+        self,
+        train_data_roots: str = None,
+        val_data_roots: str = None,
+        test_data_roots: str = None,
+        unlabeled_data_roots: str = None,
+    ) -> Dict[Subset, DatumaroDataset]:
+        """Import dataset by using Datumaro.import_from() method.
+
+        Args:
+            train_data_roots (str): Path for training data
+            val_data_roots (str): Path for validation data
+            test_data_roots (str): Path for test data
+            unlabeled_data_roots (str): Path for unlabeled data
+
+        Returns:
+            DatumaroDataset: Datumaro Dataset
+        """
+        # Construct dataset for training, validation, unlabeled
+        # TODO: currently, only MVTec dataset format can be used
+        dataset = {}
+        if train_data_roots:
+            dataset[Subset.TRAINING] = DatumaroDataset.import_from(train_data_roots, format="image_dir")
+            if val_data_roots:
+                dataset[Subset.VALIDATION] = DatumaroDataset.import_from(val_data_roots, format="image_dir")
```

The anomaly tasks currently cannot be trained without a validation set (this is a WIP). So for anomaly tasks it is mandatory that the user provides a val_data_root. Maybe we could raise an error or warning here (or elsewhere) when the `val_data_roots` is `None`, with a descriptive message explaining that anomaly tasks cannot be trained without a validation set.

Note: for anomaly tasks it is not possible to obtain a validation set by random splitting from the training set, because the validation set must contain both normal and anomalous images, while the training set contains only normal images.

---

Thanks for detailed information, I'll fix it